packages/flutter/lib/src/material_animated_icons/animated_icons.dart

```diff
@@ -12,27 +12,29 @@ part of material_animated_icons;
 /// Shows an animated icon at a given animation [progress].
 ///
 /// The available icons are specified in [AnimatedIcons].
-class AnimatedIcon extends StatelessWidget {
+class AnimatedIcon extends AnimatedWidget {
 
   /// Creates an AnimatedIcon.
   ///
   /// [progress], [color], and [icon] cannot be null.
   const AnimatedIcon({
-    @required this.progress,
+    Key key,
+    @required Animation<double> progress,
     @required this.color,
     @required this.icon,
     this.semanticLabel,
     this.textDirection,
     // TODO(amirh): add a parameter for controlling scaling behavior.
   }) : assert(progress != null),
        assert(color != null),
-       assert(icon != null);
+       assert(icon != null),
+       super(key: key, listenable: progress);
 
   /// The animation progress for the animated icon.
   /// The value is clamped to be between 0 and 1.
```

Comment:
you can't depend on this. For example, someone might use the "elastic" curve.

I recommend either extrapolating beyond 0..1, or having the code clamp the incoming values

Reply:
clamped